climada/hazard/landslide.py

```diff
@@ -188,304 +127,170 @@ def __init__(self):
+        extent.
+        Points are assigned to the gridcell they fall into, and the whole grid-
+        cell hence counts as equally affected.
+        Event frequencies are roughly estimated by 1/(total time passed in event set),
+        but such estimates are not meaningful and shouldn't be used for 
+        probabilistic calculations! Use the probabilistic method for this!
+        
+        See tutorial for details; the global ls catalog from NASA COOLR can be
+        downloaded from https://maps.nccs.nasa.gov/arcgis/apps/webappviewer/index.html?id=824ea5864ec8423fb985b33ee6bc05b7
+
         Parameters:
-            path_sourcefile (str or Path): path of file from which window should be read in
-            bbox (array): [north, east, south, west]
+        ----------
+        bbox : tuple
+            (minx, miny, maxx, maxy) geographic extent of interest
+         path_sourcefile : str
+             path to shapefile (.shp) with ls point data
+        res : float
+            resolution in degrees of the final grid cells which are created.
+            (1° ~ 111 km at the equator). 
+            Affects the hazard extent! Default is 0.00833
+        check_plots : bool
+            Whether to plot centroids, intensity & fraction of the hazard set.
+            
         Returns:
-            window_array (array): corner, width & height for Window() function of rasterio
+        --------
+            self (Landslide() inst.): instance filled with historic LS hazard
+                set for either point hazards or polygons with specified
+                surrounding extent.
         """
-        with rasterio.open(path_sourcefile) as src:
-            utm = pyproj.Proj(init='epsg:4326')  # Pass CRS of image from rasterio
-
-        lonlat = pyproj.Proj(init='epsg:4326')
-        lon, lat = (bbox[3], bbox[0])
-        west, north = pyproj.transform(lonlat, utm, lon, lat)
-
-        # What is the corresponding row and column in our image?
-        row, col = src.index(west, north)  # spatial --> image coordinates
-
-        lon, lat = (bbox[1], bbox[2])
-        east, south = pyproj.transform(lonlat, utm, lon, lat)
-        row2, col2 = src.index(east, south)
-        width = abs(col2 - col)
-        height = abs(row2 - row)
-
-        window_array = [col, row, width, height]
-
-        return window_array
-
-    def _get_raster_meta(self, path_sourcefile, window_array):
-        """get geo-meta data from raster files to set centroids adequately"""
-        raster = rasterio.open(path_sourcefile, 'r',
-                               window=Window(window_array[0], window_array[1],
-                                             window_array[2], window_array[3]))
-        pixel_width = raster.meta['transform'][0]
-        pixel_height = raster.meta['transform'][4]
+        ls_gdf_bbox = gpd.read_file(path_sourcefile, bbox=bbox)
+        
+        self.centroids.set_raster_from_pnt_bounds(bbox,res)
+
+        n_ev = len(ls_gdf_bbox)
```

Comment:
I see. This is a bit confusing then to have a variable 'bounding box' that is actually an event * locations . This can maybe be made a bit more explicit, but it is not super important.

Reply:
I think I still do not understand why a bbox would contain events and not only locations.

Reply:
it's not a bounding box. it's a geodataframe, cut to include only those entries that lie within the bounding box.I could re-name it to `gdf_cropped`or sth if that is more clar.

Reply:
yes :D